### PR TITLE
Fix sorting algorithm to work for all array types

### LINKED ISSUE #204: std.sort ignores keyF
std.sort doesn't seem to respect keyF argument:
```jsonnet
std.sort([1,2,3], keyF=function(v) error "not reached?")
```

Returns `[1, 2, 3]`, `keyF` function is not called.

### PR DESCRIPTION
**Pull request by stephenamar-db**
Fixes #204, by really calling keyF for all types in the array.

### CODE REVIEW

File: sjsonnet/src/sjsonnet/Std.scala [hunk 1]
```diff
@@ -750,8 +749,8 @@ class Std {
         case (l: Val.Obj, r: Val.Obj) =>
           val kvs = for {
             k <- (l.visibleKeyNames ++ r.visibleKeyNames).distinct
-            val lValue = Option(l.valueRaw(k, l, pos)(ev))
-            val rValue = Option(r.valueRaw(k, r, pos)(ev))
+            lValue = Option(l.valueRaw(k, l, pos)(ev))
+            rValue = Option(r.valueRaw(k, r, pos)(ev))
```
Comment (jordanmoxon-db) [inline sjsonnet/src/sjsonnet/Std.scala:753]: Does removing the immutability here fix a problem that you found?

File: sjsonnet/src/sjsonnet/Std.scala [hunk 2]
```diff
@@ -1377,39 +1376,50 @@ class Std {
...
-        )
-      case Val.Str(pos, s) => new Val.Arr(pos, s.sorted.map(c => Val.Str(pos, c.toString)).toArray)
-      case x => Error.fail("Cannot sort " + x.prettyName)
+  private def sortArr(pos: Position, ev: EvalScope, arr: Val, keyF: Val) = {
+    val vs = arr match {
+      case arr: Val.Arr => arr
+      case str: Val.Str => stringChars(pos, str.asString)
+      case _ => Error.fail("Cannot sort " + arr.prettyName)
+    }
+    if (vs.length == 0) {
```
Comment (jordanmoxon-db) [inline sjsonnet/src/sjsonnet/Std.scala:1385]: can exit early on `<= 1` right?

**Review by jordanmoxon-db — APPROVED** [review]

File: sjsonnet/src/sjsonnet/Std.scala (hunk 1 above)
Comment (stephenamar-db) [reply sjsonnet/src/sjsonnet/Std.scala:753]: it was a intellij recommendation (val is deprecated or something like taht)

File: sjsonnet/src/sjsonnet/Std.scala (hunk 2 above)
Comment (stephenamar-db) [reply sjsonnet/src/sjsonnet/Std.scala]: yup

**Review by jordanmoxon-db — APPROVED** [review]

**Pull request merged**